frontend/src/views/Lobby.tsx

```diff
@@ -69,16 +80,16 @@ function LobbyPage() {
     // Grab the user and room information; otherwise, redirect to the join page
     if (location && location.state && location.state.user && location.state.room) {
       setCurrentUser(location.state.user);
-      setRoom(location.state.room);
+      setStateFromRoom(location.state.room);
```

Comment:
Maybe instead of grabbing all of the current room information from `location`, you could just grab the `roomId` and `user` then do a GET request for the rest of the information?

I tried a variant of updating location every time the `room` variable was updated, but wasn't able to get it to work. :/ I don't know if there's another way to update the location, and a GET request seems to make more sense in terms of flow, so I think that may be a good option.

Reply:
After talking about the new design we're planning on going with, this sounds great - implemented in the latest commits.